src/lib/core/ripple/ripple-renderer.ts

```diff
@@ -164,12 +167,16 @@ export class RippleRenderer {
     // Add the ripple reference to the list of all active ripples.
     this._activeRipples.add(rippleRef);
 
+    if (!config.persistent) {
+      this._topTemporaryRipple = rippleRef;
+    }
+
     // Wait for the ripple element to be completely faded in.
     // Once it's faded in, the ripple can be hidden immediately if the mouse is released.
     this.runTimeoutOutsideZone(() => {
       rippleRef.state = RippleState.VISIBLE;
 
-      if (!config.persistent && !this._isPointerDown) {
+      if (!config.persistent && (rippleRef !== this._topTemporaryRipple || !this._isPointerDown)) {
```

Comment:
Can you add a comment that explains that we only want one ripple to be present, if the pointer is not being released yet?

Reply:
Agreed, it would be good to have a comment explaining the intent here